src/core/hle/service/jit/jit_context.cpp

```diff
@@ -350,30 +337,67 @@ class JITContextImpl {
 };
 
 void DynarmicCallbacks64::CallSVC(u32 swi) {
-    switch (swi) {
-    case 0:
+    // Service calls are used to implement helper functionality.
+    //
+    // The most important of these is the _stop helper, which transfers control
+    // from the plugin back to HLE context to return a value. However, a few more
+    // are also implemented to reduce the need for direct ARM implementations of
+    // basic functionality, like memory operations.
+    //
+    // When we receive a helper request, the swi number will be zero, and the call
+    // will have originated from an address we know is a helper function. Otherwise,
+    // the plugin may be trying to issue a service call, which we shouldn't handle.
+
+    if (swi != 0) {
+        LOG_CRITICAL(Service_JIT, "plugin issued unknown service call {}", swi);
         parent.jit->HaltExecution();
-        break;
+        return;
+    }
 
-    case 1: {
+    u64 pc{parent.jit->GetPC() - 4};
+    auto& helpers{parent.helpers};
+
+    if (pc == helpers["memcpy"] || pc == helpers["memmove"]) {
+        VAddr dest = parent.jit->GetRegister(0);
+        VAddr src = parent.jit->GetRegister(1);
+        VAddr n = parent.jit->GetRegister(2);
+
+        if (dest < src) {
+            for (size_t i = 0; i < n; i++) {
+                MemoryWrite8(dest + i, MemoryRead8(src + i));
+            }
+        } else {
+            for (ssize_t i = n - 1; i >= 0; i--) {
```

Comment:
Is ssize_t supported? Perhaps use s64 instead.
```suggestion
            for (ssize_t i = n - 1; i >= 0; i--) {
```